File: ios/QuerosermB/Core/Network/NetworkError.swift
import Foundation

enum NetworkError: LocalizedError {
    case invalidURL
    case invalidResponse
    case serverError(statusCode: Int)
    case decodingError(Error)
    case noConnection
    case unknown(Error)

    var errorDescription: String? {
        switch self {
        case .invalidURL:
            return Strings.Network.invalidURL
        case .invalidResponse:
            return Strings.Network.invalidResponse
        case .serverError(let code):
            return Strings.Network.serverError(code: code)
        case .decodingError:
            return Strings.Network.decoding
        case .noConnection:
            return Strings.Network.noConnection
        case .unknown(let error):
            return error.localizedDescription
        }
    }
}

File: android/app/src/main/java/br/com/querosermb/core/network/NetworkError.kt
package br.com.querosermb.core.network

sealed class NetworkError : Exception() {
    object InvalidURL : NetworkError()
    object InvalidResponse : NetworkError()
    data class ServerError(val statusCode: Int) : NetworkError()
    data class DecodingError(override val cause: Throwable) : NetworkError()
    object NoConnection : NetworkError()
    data class Unknown(override val cause: Throwable) : NetworkError()

    fun userMessage(): String = when (this) {
        is InvalidURL -> "URL inválida"
        is InvalidResponse -> "Resposta inválida do servidor"
        is ServerError -> "Erro no servidor (código $statusCode)"
        is DecodingError -> "Erro ao processar os dados"
        is NoConnection -> "Sem conexão com a internet"
        is Unknown -> cause.message ?: "Erro desconhecido"
    }
}
